scripts/remove_after_use/divorce_quickfiles.py

```diff
@@ -0,0 +1,178 @@
+from osf.models import QuickFolder, OSFUser, UserLog
+from osf.models.legacy_quickfiles import QuickFilesNode
+from django.core.paginator import Paginator
+from django.db.models import OuterRef, Subquery, Count
+from django.contrib.contenttypes.models import ContentType
+
+import logging
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def transfer_logs(quickfiles_node):
+    """
+    Recasts logs and saves them, old logs will be deleted with all quickfilesnodes.
+    :param quickfiles_node: a QuickFilesNode
+    :return:
+    """
+    for log in quickfiles_node.logs.all():
+        log.__class__ = UserLog
+        log.save()
+
+
+def reverse_transfer_logs(quickfiles_node):
+    """
+    Recasts logs and saves them, old logs will be deleted with all quickfilesnodes.
+    :param quickfiles_node: a QuickFilesNode
+    :return:
+    """
+    for log in quickfiles_node.logs.all():
+        log.__class__ = NodeLog
+        log.save()
+
+
+def create_quickfolders():
+    """
+    Bulk creates a Quickfolder for every user.
+    :return:
+    """
+    users = OSFUser.objects.all()
+    user_content_type_id = ContentType.objects.get_for_model(OSFUser).id
+    quickfolder_content_type_id = ContentType.objects.get_for_model(QuickFolder).id
+
+    paginated_users = Paginator(users, 1000)
```

Comment:
I wouldn't use Paginator - it adds a lot of overhead trying to break all the users into pages of 1000 users each (this took 15 min) - I'd explore just looping through users.